cpp/memilio/utils/logging.h

```diff
@@ -127,38 +127,24 @@ inline void log(LogLevel level, spdlog::string_view_t fmt, const Args&... args)
 
 } // namespace mio
 
-/**
- * @brief The fmt::formatter class provides formatting capabilities for the ad::gt1s<double>::type 
- * to help SPD log to handle it.
- */
-template <>
-struct fmt::formatter<ad::gt1s<double>::type> {
-    constexpr auto parse(format_parse_context& ctx) -> decltype(ctx.begin())
-    {
-        return ctx.end();
-    }
-    template <typename FormatContext>
-    auto format(const ad::gt1s<double>::type& input, FormatContext& ctx) -> decltype(ctx.out())
-    {
-        return format_to(ctx.out(), "{}", ad::value(input));
-    }
-};
```

Comment:
don't see why so much became superfluous?!

Reply:
Two things. First, since we only print the value part of the AD type, I use a simpler part of the [fmt API](https://fmt.dev/latest/api/#formatting-user-defined-types), i.e. format_as,  to do essentially the same as this struct in fewer lines.
Second, I used the same template deduction @lenaploetzke introduced in #1155 to handle both (probably all) AD types, hence there is only one function replacing both types.